hoprd/db/api/src/aliases.rs

```diff
@@ -40,53 +44,75 @@ impl HoprdDbMetadataOperations for HoprdDb {
+                    "own alias cannot be modified".into(),
+                ));
+            }
+        }
+
         let new_aliases = aliases
             .iter()
             .map(|entry| hoprd_db_entity::aliases::ActiveModel {
+                id: Default::default(),
                 peer_id: sea_orm::ActiveValue::Set(entry.peer_id.clone()),
                 alias: sea_orm::ActiveValue::Set(entry.alias.clone()),
-                ..Default::default()
             })
             .collect::<Vec<_>>();
 
         let _ = hoprd_db_entity::aliases::Entity::insert_many(new_aliases)
+            .on_conflict(
+                OnConflict::columns([Column::PeerId])
+                    .update_column(Column::PeerId)
+                    .to_owned(),
+            )
             .exec(&self.metadata)
             .await?;
 
         Ok(())
     }
 
     async fn set_alias(&self, peer: String, alias: String) -> Result<()> {
-        // update the entry if peer already in the db
-        let row = hoprd_db_entity::aliases::Entity::find()
-            .filter(hoprd_db_entity::aliases::Column::PeerId.eq(peer.clone()))
-            .one(&self.metadata)
-            .await?;
-
-        if let Some(model) = row {
-            self.delete_alias(model.alias).await?;
+        if let Some(me) = self.resolve_alias(ME_AS_ALIAS.to_string()).await.unwrap() {
+            if me == peer {
+                return Err(crate::errors::DbError::LogicalError(
+                    "own alias cannot be modified".into(),
+                ));
+            }
         }
-
-        let new_alias = hoprd_db_entity::aliases::ActiveModel {
-            peer_id: sea_orm::ActiveValue::Set(peer.clone()),
-            alias: sea_orm::ActiveValue::Set(alias.clone()),
-            ..Default::default()
-        };
-        let _ = hoprd_db_entity::aliases::Entity::insert(new_alias)
-            .exec(&self.metadata)
-            .await?;
+        let _ = self
+            .metadata
+            .transaction::<_, (), DbErr>(|tx| {
+                Box::pin(async move {
+                    let row = hoprd_db_entity::aliases::Entity::find()
+                        .filter(hoprd_db_entity::aliases::Column::PeerId.eq(peer.clone()))
+                        .one(tx)
+                        .await?
+                        .map(|model| model.id);
+
+                    let new_alias = hoprd_db_entity::aliases::ActiveModel {
+                        peer_id: sea_orm::ActiveValue::Set(peer.clone()),
+                        alias: sea_orm::ActiveValue::Set(alias.clone()),
+                        id: if let Some(id) = row {
+                            sea_orm::ActiveValue::Set(id)
+                        } else {
+                            sea_orm::ActiveValue::NotSet
+                        },
+                    };
+
+                    new_alias.save(tx).await?;
+
+                    Ok(())
+                })
+            })
+            .await;
```

Comment:
Don't you want to test via `?` to see that the tx didn't fail?

Or you may wanna return directly the `Result` you got by `await`ing the `transaction` function.